AAXtoMP3

```diff
@@ -510,6 +517,18 @@ save_metadata() {
     # Then we delete all ':' since they make a filename invalid.
     jq -r '.content_metadata.chapter_info.chapters[] | "Chapter # start: \(.start_offset_ms/1000), end: \((.start_offset_ms+.length_ms)/1000) \n#\n# Title: \(.title)"' "${extra_chapter_file}" \
       | tr -d ':' >> "$metadata_file"
+    # In case we want to use a single file m4b we need to extract the
+    # chapter titles from the .json generated by audible–cli and store
+    # them correctly formatted for mp4chaps in a chapter.txt
+    if [ "${mode}" == "single" ]; then
+      jq -r \
+        'def pad(n): tostring | if (n > length) then ((n - length) * "0") + . else . end;
+        .content_metadata.chapter_info.chapters |
+        to_entries |
+        .[] |
+        "CHAPTER\((.key | pad(2)))=\(((.value.start_offset_ms / (1000*60*60)) %24 | floor | pad(2))):\(((.value.start_offset_ms / (1000*60)) %60 | floor | pad(2))):\(((.value.start_offset_ms / 1000) %60 | floor | pad(2))).\((.value.start_offset_ms % 1000 | pad(3)))
```

Comment:
The padding for the chapter number is not necessary
```suggestion
        "CHAPTER\((.key))=\(((.value.start_offset_ms / (1000*60*60)) %24 | floor | pad(2))):\(((.value.start_offset_ms / (1000*60)) %60 | floor | pad(2))):\(((.value.start_offset_ms / 1000) %60 | floor | pad(2))).\((.value.start_offset_ms % 1000 | pad(3)))
```

Reply:
ah, good to know. Even better :)